src/kixi/datastore/metadatastore/elasticsearch.clj

```diff
@@ -1,184 +1,74 @@
-                                   ::ms/provenance {:type "nested"
-                                                    :properties {::ms/source string-stored-not_analyzed
-                                                                 :kixi.user/id string-stored-not_analyzed
-                                                                 ::ms/pieces-count {:type "integer"}
-                                                                 ::ms/parent-id string-stored-not_analyzed}}
-                                   ::ms/segmentation {:type "nested"
-                                                      :properties {::seg/type string-stored-not_analyzed
-                                                                   ::seg/line-count {:type "integer"}
-                                                                   ::seg/value string-stored-not_analyzed}}
-                                   ::ms/sharing {:type "nested"
-                                                 :properties (zipmap ms/activities
-                                                                     (repeat string-stored-not_analyzed))}})}}]
-    (esi/create conn index-name {:mappings mapping-types
-                                 :settings {}})))
-
-(defn ensure-index
-  [conn]
-  (when-not (esi/exists? conn index-name)
-    (build-index conn)))
-
-(defn- get-document
-  [conn id]
-  (esd/get conn
-           index-name 
-           doc-type 
-           id
-           {:preference "_primary"}))
+(def doc-def
+  {::ms/id string-stored-not_analyzed
+   ::ms/type string-stored-not_analyzed
+   ::ms/name {:type "string"
+              :store "yes"}
+   ::ss/id string-stored-not_analyzed
+   ::ms/provenance {:type "nested"
+                    :properties {::ms/source string-stored-not_analyzed
+                                 :kixi.user/id string-stored-not_analyzed
+                                 ::ms/pieces-count {:type "integer"}
+                                 ::ms/parent-id string-stored-not_analyzed}}
+   ::ms/segmentation {:type "nested"
+                      :properties {::seg/type string-stored-not_analyzed
+                                   ::seg/line-count {:type "integer"}
+                                   ::seg/value string-stored-not_analyzed}}
+   ::ms/sharing {:type "nested"
+                 :properties (zipmap ms/activities
+                                     (repeat string-stored-not_analyzed))}})
 
 (s/fdef update-metadata-processor
         :args (s/cat :conn #(instance? clojurewerkz.elastisch.rest.Connection %)
                      :update-req ::cs/file-metadata-updated))
 
+(def merge-data
+  (partial es/merge-data index-name doc-type))
+
+(def cons-data
+  (partial es/cons-data index-name doc-type))
+
+(def get-document
+  (partial es/get-document index-name doc-type))
+
+(def get-document-key
+  (partial es/get-document-key index-name doc-type))
+
+(def present?
```

Comment:
It occurs to me that `exists?` is a more common term (regardless of what elastisch uses), if this is in fact checking for the presence of a document